### PR TITLE
[knx] Allow receiving DPT 235.001

### LINKED ISSUE #15159: [knx] Add support for DPT 235.001 for energy with tariff data point
At least Hager TE331 will provide an objet for the current electrical energy associated with tariff to handle subscription with multiple tariff in France (a.k.a. EDF Tempo).

Specification of this datapoint format is provided on the following image : [DPT235.001](https://ibb.co/7NrgKyj)

There are 4-bytes for the current counter value (integer in Wh, always increasing)
There is 1 byte for the current tariff (in reality, a number between 0 and 10)
There is 1 byte for control

We need to be able to extract both values independently to be able to store them in the long term for graphing.

If needed, I can provide KNX diagnostics with examples values.

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Composed type not yet supported by Calimero, thus only receiving data is implemented.
Configure DPT 235.001 to receive ActiveEnergy.
Configure DPT 235.61001 to receive Tariff information.

Fixes #15159

### CODE REVIEW

**Comment by holgerfriedrich** [comment]
@dsacchet FYI

File: bundles/org.openhab.binding.knx/README.md [hunk 1]
```diff
@@ -267,6 +267,18 @@ With `*-control` channels, the state is not owned by any device on the KNX bus,
...
+
+OpenHAB supports all DPTs supported by the corresponding release of Calimero library.
+
+Additional DPTs have been introduced to add functionality:
+
+| DPT           | Description                                                 | Remark     |
+|---------------|-------------------------------------------------------------|------------|
+| DPT 232.60000 | DPT 232.600 with HSB instead of RGB data (see below)        | read/write |
+| DPT 235.001   | Composed DPT 235.001, first element ActiveEnergy (Wh)       | read only  |
+| DPT 235.61001 | Composed DPT 235.001, second element Tariff (plain number)  | read only  |
```
Comment (J-N-K) [inline bundles/org.openhab.binding.knx/README.md:280]: 235.001 is the defined DPT, correct? We added 00 as custom type for 232.600, so maybe we should use 235.00100 here. If we need another one, we could then increment the last digit.

Comment (holgerfriedrich) [reply bundles/org.openhab.binding.knx/README.md:280]: I thought about a good naming scheme as well, as more of those multipart DPTs may be needed in the future.

My idea for the subtypes is as follows:

- 60000 is base for extensions, this is in the area of manufacturer specific DPTs, .
- the second digit refers to the element of the multipart message
- digits 3-5 are the old subtype (as there are subtypes like .600 we need 3 digits).

The original DPT is kept for the main content if the multipart message (in this case ActiveEnergy).

232.60000 does not follow this scheme, but has been introduced before.

**Comment by lsiepel** [comment]
Code lgtm, before a merge, i think it would be good to have a comment from @kaikreuzer  or @J-N-K  about this review comment: https://github.com/openhab/openhab-addons/pull/16094#discussion_r1441722226

While tests are supplied, did you also have test results from @dsacchet ?

**Comment by holgerfriedrich** [comment]
No test results were reported. Bit I have added tests with raw frames on byte level, and did some manual tests with ETS. Should be good enough to go....

File: bundles/org.openhab.binding.knx/README.md (hunk 1 above)
Comment (J-N-K) [reply bundles/org.openhab.binding.knx/README.md:280]: I'm fine with the explanation.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**